### PR TITLE
move checking existance of idn_to_ascii function

### PR DESCRIPTION
**Pull request by kriptonuz**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #120

### CODE REVIEW

**Review by vjik — APPROVED** [review]

File: src/Rule/Url.php [hunk 1]
```diff
@@ -108,6 +101,11 @@ public function pattern(string $pattern): self
     public function enableIDN(): self
     {
         $new = clone $this;
+
+        if (!function_exists('idn_to_ascii')) {
+            throw new \RuntimeException('In order to use IDN validation intl extension must be installed and enabled.');
+        }
```
Comment (viktorprogger) [inline src/Rule/Url.php:107]:
```suggestion
        if (!function_exists('idn_to_ascii')) {
            throw new \RuntimeException('In order to use IDN validation intl extension must be installed and enabled.');
        }

        $new = clone $this;
```
There is no need to clone the instance if `idn_to_ascii` doesn't exist.

**Review by viktorprogger — CHANGES_REQUESTED** [review]
This behavior should be checked within a test. Here is an example [how to mock the `function_exists` function](https://stackoverflow.com/questions/32007032/is-it-possible-to-mock-the-non-existence-of-a-function-in-phpunit#answer-34386422).

**Review by samdark — APPROVED** [review]
Good change but tests should be added for it.

File: tests/Rule/UrlTest.php [hunk 2]
```diff
@@ -13,6 +25,13 @@
  */
 class UrlTest extends TestCase
 {
+    public static bool $idnFunctionException = false;
```
Comment (samdark) [inline tests/Rule/UrlTest.php:28]: Why do we need it if it's not used anywhere?

**Comment by samdark** [comment]
👍

**Pull request merged**